File: frontend/src/types/place.ts
export interface PlaceTypeDto {
  id: string;
  name: string;
  description: string;  
  createdAt: string;
}

export interface PlaceDto {
  id: string;
  name: string;
  description: string;
  placeTypeId: string;
  placeType?: PlaceTypeDto;
  latitude: number;
  longitude: number;
  address?: string;
  imageId?: string;
  creatorId: string;
  creatorName: string;
  createdAt: string;
  updatedAt?: string;
  status: string;
  
  // UI-specific / Additional fields
  reviewerId?: string;
}

/** A place that hasn't been created in the DB yet — deferred until event creation. */
export interface PendingPlace {
  pending: true;
  name: string;
  description: string;
  placeTypeId: string;
  latitude: number;
  longitude: number;
  creatorId: string;
  address?: string;
}

/** Either an existing place from the DB or a new place pending creation. */
export type SelectedPlace = PlaceDto | PendingPlace;

export function isPendingPlace(place: SelectedPlace): place is PendingPlace {
  return "pending" in place && place.pending === true;
}
export type Place = PlaceDto;
export type PlaceType = PlaceTypeDto;

File: DomainLayer/Entities/Place.cs
using DomainLayer.Common;
using DomainLayer.Entities.Enums;
using System.ComponentModel.DataAnnotations.Schema;

namespace DomainLayer.Entities
{
    public class Place : BaseEntity
    {
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        
        [ForeignKey(nameof(PlaceType))]
        public Guid PlaceTypeId { get; set; }
        public double Latitude { get; set; }
        public double Longitude { get; set; }
        public string? Address { get; set; }
        public Guid? ImageId { get; set; }
        public Guid CreatorId { get; set; }
        public StatusType Status { get; set; } = StatusType.Pending;
        public Guid? ReviewerId { get; set; }

        public PlaceType? PlaceType { get; set; }
        public User Creator { get; set; } = null!;
        public User? Reviewer { get; set; }
    }
}